io-engine/src/bdev/nexus/nexus_channel.rs

```diff
@@ -239,22 +253,35 @@ impl<'n> NexusChannel<'n> {
                         );
                         writers.push(hdl);
                     }
-                    Err(e) => {
-                        c.set_faulted_state(FaultReason::CantOpen);
-                        error!(
-                            "{self:?}: failed to get I/O handle \
-                                for {c:?}: {e}"
-                        );
-                    }
+                    Err(e) => match e {
+                        CoreError::OpenBdev {
+                            source: _,
+                        } => ret = Errno::EAGAIN as i32,
+                        _ => {
+                            c.set_faulted_state(FaultReason::CantOpen);
+                            error!(
+                                "{self:?}: failed to get I/O handle \
+                                            for {c:?}: {e}"
+                            );
+                        }
+                    },
                 });
         }
 
+        if ret == Errno::EAGAIN as i32 {
+            // This would mean that we'll re-attempt to get the block device
+            // handles for all the child devices again, and not only
+            // the one for which we received EAGAIN.
+            return ret;
```

Comment:
I think there's an issue here, this might have succeeded for core 0 and not for other cores at which point we have a different view of children among cores which seems dangerous.

Reply:
Agree, but it's no worse than what's possible today as well because the spdk callback to reconfigure for various spdk threads and cores are asynchronously and serially dispatched during channel iteration. Those callbacks can lag in time and land into similar situation.

Reply:
Well currently we'd assert and crash AIUI which is in a way safer.
I think we should attempt to pre-connect the handles before reconfigure, and then on reconnect we could just "acquire" those handles?